src/MOI_wrapper.jl

```diff
@@ -24,16 +22,35 @@ mutable struct DSDPSolverInstance <: SDOI.AbstractSDSolverInstance
     z_computed::Bool
 
     options::Dict{Symbol,Any}
-    function DSDPSolverInstance(; kwargs...)
-        m = new(C_NULL, C_NULL, 0, Int[], Int[], C_NULL, 0, Int[], Int[], Cdouble[],
-            true, true, Cdouble[], true, Dict{Symbol, Any}(kwargs))
-        finalizer(m, _free)
-        m
+    function SDOptimizer(; kwargs...)
+        optimizer = new(C_NULL, C_NULL, 0, Int[], Int[], C_NULL, 0, Int[],
+                        Int[], Cdouble[], true, true, Cdouble[], true,
+                        Dict{Symbol, Any}(kwargs))
+        finalizer(_free, optimizer)
+        return optimizer
     end
 end
-DSDPInstance(; kws...) = SDOI.SDOIInstance(DSDPSolverInstance(; kws...))
+Optimizer(; kws...) = SDOI.SDOIOptimizer(SDOptimizer(; kws...))
+
+MOI.get(::SDOptimizer, ::MOI.SolverName) = "DSDP"
+
+function MOI.empty!(optimizer::SDOptimizer)
+    _free(optimizer)
+    optimizer.nconstrs = 0
+    optimizer.blkdims = Int[]
+    optimizer.blk = Int[]
+    optimizer.nlpdrows = 0
+    optimizer.lpdvars = Int[]
+    optimizer.lpdrows = Int[]
+    optimizer.lpcoefs = Cdouble[]
+
+    optimizer.x_computed = true
+    optimizer.y_valid = true
```

Comment:
Preexisting: Are these default values for ``x_computed`` and ``y_valid`` reasonable for an empty model?

Reply:
Looping back here after reading the code, my belief is that ``x_computed=false`` and ``y_valid=true`` are correct defaults.